layout/AddressesUsingIdsQuery.js

```diff
@@ -193,6 +197,11 @@ class AddressesUsingIdsQuery extends Query {
       base.query.function_score.query.bool.should.push(createAddressShould(vs));
     }
 
+    // default address:street:slop to 0 if unset
+    if (!vs.isset('address:street:slop')) {
+      vs.var('address:street:slop', 0);
```

Comment:
@missinglink, I put this here as a shortcut. I notice that most of our code in API makes copies of the variable store rather than changing variables in views. Do you think that would be a better approach here?

Reply:
I would prefer that if the variable was unset then we totally omit it from the query, rather than trying to set it to the default in ES.

The default slop set by elasticsearch is `0` but this could change at any time.

That's how most of the views I've written work, they have the 'mandatory' properties in the main blob and then a bunch of if conditions below which set optional params.

Reply:
If you want to set defaults then the correct place to do this would be in the `search_defaults` file in API.

Reply:
Eg. https://github.com/pelias/query/pull/99

Reply:
Funny, I prefer the opposite, I explicitly wrote it intending to ensure the slop parameter is always in the query so that it's clear what its set to. As far as pelias/query is concerned, I believe the default should be set to zero. If Elasticsearch changes the default slop value that would be a _huge_ pain and I'd prefer we avoid that (extremely slim) possibility.

The default in `search_defaults` in the API is going to be 1 to get the behavior we want, and that is handled in https://github.com/pelias/api/pull/1322

Reply:
As a general principal I think that we should reserve the option of omission of optional properties.

There is a valid case for a developer wishing to rely on the default behaviour rather than specifying it explicitly.

I can see the benefit of setting it to 0, mainly for clarity (at the cost of verbosity).

Although I feel it is still better as an **optional** parameter where the caller (pelias/api) is able to make the decision about whether it will be included or excused from the query.

Just my 2c :)

Reply:
Funny thing is that our existing phrase view does it as you suggest:

https://github.com/pelias/query/blob/master/view/phrase.js

Reply:
It seems odd to me in the existing `phrase.js` view that we have two optional properties (`slop` and `cutoff_frequency`) which are handled differently.

I think we agree that we wouldn't want to set a default value for `cutoff_frequency` `fuzziness` et al?

There is something about `slop` which makes it feel like it's mandatory even though it isn't.